### PR TITLE
Reconfigure retries & timeouts for typical ingestion DAGs

### LINKED ISSUE #348: Use `execution_timeout` rather than `dagrun_timeout`
## Description

Currently we're using `dagrun_timeout` to set a timeout for our provider workflows. According to [this documentation](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/index.html), the `dagrun_timeout` allows one to:

> specify how long a DagRun should be up before timing out / failing, so that new DagRuns can be created. The timeout is only enforced for scheduled DagRuns.

It seems like that second sentence may not be true for v2.2.3, since we saw an instance where a manually triggered DAG run also (inappropriately) respected the `dagrun_timeout`. The `dagrun_timeout` is a bit confusing, and [has been discussed on the dev mailing list before](https://lists.apache.org/thread/h70wnb0jqod6wbxml5bvbtj5018sh3mj).

Additionally it looks like DAGs that hit the `dagrun_timeout` _skip_ rather than failing, which then doesn't trigger our failure alerting.

What we really want is [`execution_timeout` at the task level](https://airflow.apache.org/docs/apache-airflow/stable/concepts/tasks.html#timeouts). This would allow us to have fine-grain control of task execution _and_ set the tasks to failed rather than skipped once they're run (thus triggering failure alerting).

## Additional context

We should probably avoid using `dagrun_timeout` altogether, but this is a priority for the provider DAGs.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #349: Improve provider workflow retries
## Description

Presently the provider DAG factory defaults to 4 (total) retries per task:

https://github.com/WordPress/openverse-catalog/blob/49d30ed1043b566350ebf4c8645feefc823675f6/openverse_catalog/dags/common/dag_factory.py#L30

We should make a few adjustments here:

- Retries should be set to `0` for the data pull steps so that if a provider fails, we want it to 1) stop trying so we can investigate and don't immediately run into the same issue and 2) ingest what we've been able to pull up to that point.
- Global retries should be knocked down to `2` (meaning `3` total attempts).
- Retry delay should be set to 5 minutes rather than 15. Currently 4 retries after 15 minutes means that we won't get error reporting until a full hour after the task was initially started. This feedback loop should be a bit tighter & shorter (while still allowing for retries). 

## Additional context

This work will likely need to come after #285 since the task flow will be modified by that PR.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #348 by @AetherUnbound, fixes #349 by @AetherUnbound

## Description

This PR tweaks a few settings regarding task retries and timeouts for the typical ingestion DAGs. By "typical ingestion DAGs", I mean the dags that are run daily or monthly and ingest new data, either from a particular date range or for all of time.

Retries for all ingestion DAG related task have been reduced from 4 attempts total to 3. The time between retries has also been reduced from 15 minutes to 5 minutes.

Retries for the data pull steps have been disabled completely. This is to prevent cases where an ingestion fails, retries, gets to the same place (usually hours later) only to fail again. Failing immediately gives the subsequent loading steps the opportunity to ingest data that's already been collected, it notifies the team via Slack, and prevents unnecessary API calls to the provider.

I've changed the `dagrun_timeout` variable to the `execution_timeout` variable. More discussion on this can be found in #348, but the short of it is that `execution_timeout` gives us granular control on the data pull task, and allows the opportunity to ingest data that's already been collected even if the task times out. I've also added a base `execution_timeout` of 1 hour to all other tasks within the DAG. This will prevent the loader steps from hanging, potentially indefinitely. With these guards in place, a DAG should never run more than a few hours over the `execution_timeout` of the ingestion task. We can also further tighten this loop by increasing the maximum number of DAGs which can run at once while constraining the ingestion tasks to use [pools](https://airflow.apache.org/docs/apache-airflow/stable/concepts/pools.html).

## Testing Instructions

Most of these changes are likely best observed in production. If you want to test the retry changes, alter a provider script to fail and then observe that it fails and loads any available data immediately.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/common/dag_factory.py [hunk 1]
```diff
@@ -158,7 +158,10 @@ def create_provider_api_workflow(
             python_callable=_push_output_paths_wrapper,
             op_kwargs=pull_kwargs,
             depends_on_past=False,
-            execution_timeout=dagrun_timeout if dated else None,
```
Comment (stacimc) [inline openverse_catalog/dags/common/dag_factory.py:161]: It seems reasonable to me that we'd want the timeout across the board, but I'm curious why it was previously applied only for dated data pulls? 🤔

Comment (AetherUnbound) [reply openverse_catalog/dags/common/dag_factory.py:161]: I'm not sure! This is [how it was prior to my interactions with the codebase](https://github.com/WordPress/openverse-catalog/blob/7ee62451343ee04a895290b23df9d9c79703dc74/openverse_catalog/dags/util/dag_factory.py#L77-L81) and I wanted to preserve behavior when I refactored this.

Comment (stacimc) [reply openverse_catalog/dags/common/dag_factory.py:161]: Mysterious! 😄 At any rate I don't see any harm in adding the timeout to the undated runs. Timeouts can always be adjusted.

**Review by stacimc — APPROVED** [review]
This looks good as far as my understanding goes 😄 Thank you for the excellent description, your reasoning seems sound to me! I tested the retry changes locally and all looks well 👍

File: openverse_catalog/dags/common/dag_factory.py (hunk 1 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/common/dag_factory.py:161]: Exactly! I think while we're activating the DAGs, we'll probably need to go back and adjust some of the timeouts (for instance, I know that Freesound will likely take quite a while to process).

Comment (zackkrida) [reply openverse_catalog/dags/common/dag_factory.py:214]: This makes sense to me too FWIW 😄

**Review by zackkrida — APPROVED** [review]
lgtm! thanks for explaining the motivation so well and adding new documentation. there's fewer code changes here than I anticipated.

**Pull request merged**